pandas/core/series.py

```diff
@@ -4042,7 +4045,13 @@ def _try_cast(arr, take_fast_path):
 
         try:
             subarr = maybe_cast_to_datetime(arr, dtype)
-            if not is_extension_type(subarr):
+            # Take care in creating object arrays (but iterators are not
+            # supported):
+            if is_object_dtype(dtype) and (is_list_like(subarr) and
```

Comment:
this is pretty hard to read, but ok for now, see if can simplify in the future

Reply:
Yes, for sure we will need some unified mechanism to process iterators